src/notation/internal/notationinteraction.cpp

```diff
@@ -405,29 +405,57 @@ mu::engraving::ShadowNote* NotationInteraction::shadowNote() const
     return score()->shadowNote();
 }
 
-bool NotationInteraction::showShadowNote(const PointF& pos)
+void NotationInteraction::showShadowNoteForPosition(const PointF& pos)
 {
     const mu::engraving::InputState& inputState = score()->inputState();
     mu::engraving::ShadowNote& shadowNote = *score()->shadowNote();
 
     ShadowNoteParams params;
     if (!score()->getPosition(&params.position, pos, inputState.voice())) {
         shadowNote.setVisible(false);
-        m_shadowNoteChanged.notify();
-        return false;
+        m_shadowNoteChanged.send(/*visible*/ false);
+        return;
     }
 
     params.duration = inputState.duration();
     params.accidentalType = inputState.accidentalType();
     params.articulationIds = inputState.articulationIds();
 
-    const bool show = showShadowNote(shadowNote, params);
-    m_shadowNoteChanged.notify();
+    const bool show = doShowShadowNote(shadowNote, params);
+    m_shadowNoteChanged.send(show);
+}
+
+void NotationInteraction::showShadowNoteForMidiPitch(const uint8_t pitch)
+{
+    const mu::engraving::InputState& inputState = score()->inputState();
+
+    Segment* inputSegment = inputState.segment();
+    const Staff* inputStaff = inputState.staff();
+    const Fraction tick = inputSegment->tick();
+    if (!inputSegment || !inputStaff || !tick.isValid() || inputStaff->isDrumStaff(tick) || inputStaff->isTabStaff(tick)) {
+        //! NOTE: Not yet compatible with drum or tab staves...
+        hideShadowNote();
+        return;
+    }
+
+    const double mag = inputStaff->staffMag(tick);
+    const double lineDist = inputStaff->staffType(tick)->lineDistance().val()
+                            * (inputStaff->isTabStaff(tick) ? 1 : .5)
+                            * mag
+                            * score()->style().spatium();
 
-    return show;
+    const int octave = pitch / 12;
+    const int line = octave * 7 + mu::engraving::pitch2step(pitch);
+    const ClefType clef = inputStaff->clef(tick);
+    const int rLine = mu::engraving::relStep(line, clef);
+
+    const double xPos = inputSegment->canvasPos().x();
+    const double yPos = inputSegment->system()->staffCanvasYpage(inputStaff->idx()) + rLine * lineDist;
+
+    showShadowNoteForPosition({ xPos, yPos });
```

Comment:
If possible, it would be great if we could directly call `doShowShadowNote` here, to avoid semi-expensively calculating a position from segment/staff/line, only to semi-expensively calculate basically a segment/staff/line from that position.